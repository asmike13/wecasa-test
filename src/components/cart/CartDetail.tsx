import { ICartItem } from 'src/redux/reducers/cartSlice'

export interface ICartDetailProps {
	items: ICartItem[]
	onRemoveOne: (item: ICartItem) => void
	onAddOne: (item: ICartItem) => void
	onRemoveAll: (item: ICartItem) => void
}

const CartDetail = ({
	items,
	onRemoveOne,
	onAddOne,
	onRemoveAll,
}: ICartDetailProps) => {
	return (
		<div className='cart-detail'>
			{items.map((item) => (
				<div className='cart-detail-item' key={item.prestation.reference}>
					<span>
						{`${item.prestation.title} ${item.category.title}`}
					</span>
					<div className='cart-detail-item-update'>
						<button onClick={() => onRemoveOne(item)}>-1</button>
						{`x ${item.prestation.quantity}`}
						<button onClick={() => onAddOne(item)}>+1</button>
						<button onClick={() => onRemoveAll(item)}>Remove</button>
					</div>
				</div>
			))}
		</div>
	)
}

export default CartDetail
