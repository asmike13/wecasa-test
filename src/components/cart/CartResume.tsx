import './styles.scss'

export interface ICartResumeProps {
	hasItems: boolean
	totalTime: string
	totalPrice: string
}

const CartResume = ({
	hasItems,
	totalTime,
	totalPrice,
}: ICartResumeProps) => {
	return (
		<>
			{hasItems
				? <>
					<span>Panier :</span>
					<ul>
						<li>{`Prix total : ${totalPrice}`}</li>
						<li>{`Dur√©e totale : ${totalTime}`}</li>
					</ul>
				</>
				: 'Panier vide'
			}
		</>
	)
}

export default CartResume
