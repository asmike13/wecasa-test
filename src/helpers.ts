export const timeFormat = (timeInMinutes: number) => {
	const hours = Math.floor(timeInMinutes / 60)
	const minutes = timeInMinutes % 60

	const displayHours = hours > 0 ? `${hours}H` : ''
	const displayMinutes = minutes > 0 ? `${minutes}min` : ''

	return `${displayHours} ${displayMinutes}`
}

export const priceFormat = (priceInCents: number) => {
	const price = Math.floor(priceInCents / 100)
	const cents = priceInCents % 100

	const displayPrice = price > 0 ? price : ''
	const displayCents = cents > 0 ? `.${cents}` : ''

	return `${displayPrice}${displayCents}â‚¬`
}
