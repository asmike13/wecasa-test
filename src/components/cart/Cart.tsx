import ActionButton from '../buttons/ActionButton'
import CartDetail, { ICartDetailProps } from './CartDetail'
import CartResume, { ICartResumeProps } from './CartResume'
import './styles.scss'

export type TCartProps = {
	cartResumeProps: ICartResumeProps
	cartDetailProps: ICartDetailProps
	height: string
	showDetail: boolean
	setShowDetail: (show: boolean) => void
}

const Cart = ({
	cartResumeProps,
	cartDetailProps,
	height,
	showDetail,
	setShowDetail,
}: TCartProps) => {
	return (
		<>
			<div className='cart-margin' style={{ height }} />
			<div className='cart'>
				<CartResume {...cartResumeProps} />
				{cartResumeProps.hasItems &&
					<ActionButton setAction={() => setShowDetail(!showDetail)} value={showDetail ? 'Masquer détail' : 'Voir détail'} />
				}
				{cartResumeProps.hasItems && showDetail &&
					<CartDetail {...cartDetailProps} />
				}
			</div>
		</>
	)
}

export default Cart
