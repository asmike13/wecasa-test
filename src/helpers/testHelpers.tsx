import { Provider } from 'react-redux'
import { store } from 'src/store'

export const StoreWrapper = (props: any) => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
