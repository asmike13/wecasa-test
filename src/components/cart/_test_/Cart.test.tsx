import renderer from 'react-test-renderer';
import { StoreWrapper } from 'src/helpers/testHelpers';
import Cart, { TCartProps } from '../Cart';
import { props as cartResumeProps } from './CartResume.test'
import { props as cartDetailProps } from './CartDetail.test'

export const props: TCartProps = {
	cartResumeProps,
	cartDetailProps,
	height: '10px',
	showDetail: false,
	setShowDetail: jest.fn(),
}

it('should render Cart as snapshot without detail', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<Cart {...props} />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})

it('should render Cart as snapshot with detail', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<Cart {...props} showDetail />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})
