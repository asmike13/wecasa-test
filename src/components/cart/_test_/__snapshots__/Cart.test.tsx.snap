// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Cart as snapshot with detail 1`] = `
Array [
  <div
    className="cart-margin"
    style={
      Object {
        "height": "10px",
      }
    }
  />,
  <div
    className="cart"
  >
    Panier vide
  </div>,
]
`;

exports[`should render Cart as snapshot without detail 1`] = `
Array [
  <div
    className="cart-margin"
    style={
      Object {
        "height": "10px",
      }
    }
  />,
  <div
    className="cart"
  >
    Panier vide
  </div>,
]
`;

exports[`should render CartDetail as snapshot with items 1`] = `
<div
  className="cart-detail"
>
  <div
    className="cart-detail-item"
  >
    <span>
      pTitle cTitle
    </span>
    <div
      className="cart-detail-item-update"
    >
      <button
        onClick={[Function]}
      >
        -1
      </button>
      x 10
      <button
        onClick={[Function]}
      >
        +1
      </button>
      <button
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
  <div
    className="cart-detail-item"
  >
    <span>
      pTitle2 cTitle2
    </span>
    <div
      className="cart-detail-item-update"
    >
      <button
        onClick={[Function]}
      >
        -1
      </button>
      x 3
      <button
        onClick={[Function]}
      >
        +1
      </button>
      <button
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;

exports[`should render CartDetail as snapshot without items 1`] = `
<div
  className="cart-detail"
/>
`;

exports[`should render CartResume as snapshot with items 1`] = `
Array [
  <span>
    Panier :
  </span>,
  <ul>
    <li>
      Prix total : 100€
    </li>
    <li>
      Durée totale : 1H 4min
    </li>
  </ul>,
]
`;

exports[`should render CartResume as snapshot without items 1`] = `"Panier vide"`;
